src/main/scala/piconot/internal/Empty.scala

```diff
@@ -0,0 +1,66 @@
+package piconot.internal
+
+import PicoBotExtender._
+
+import java.io.File
+import scalafx.application.JFXApp
+
+import picolib.maze.Maze
+import picolib.semantics._
+
+/**
+ *  This uses our internal DSL syntax to make the rules for picobot
+ */
+
+object EmptyRoomDSL extends JFXApp {
+  val emptyMaze = Maze("resources" + File.separator + "empty.txt")
+
+  val rules : List[Rule] = List(
+    
+    /////////////////////////////////////////////////////////
+    // State 0: go West
+    /////////////////////////////////////////////////////////
+
+    // as long as West is unblocked, go West
+    "0W" -> "go W changeState 0",
+
```

Comment:
This is definitely pushing the boundary of what an internal DSL is. While it adds a bit of extra semantics that help, the string parsing pretty much makes it external.